Core/MIPS/x86/X64IRCompALU.cpp

```diff
@@ -130,8 +134,13 @@ void X64JitBackend::CompIR_Assign(IRInst inst) {
 		break;
 
 	case IROp::Ext8to32:
+		regs_.Map(inst);
+		MOVZX(32, 8, regs_.RX(inst.dest), regs_.R(inst.src1));
```

Comment:
I realized in #17948 that we need to be smarter about this.  In theory we could use `regs_.MapWithExtra(inst, { { 'G', inst.src1, 1, MIPSMap::INIT | X64Map::LOW_SUBREG } });`, but that's pretty ugly.

I think it'd be reasonable to do `regs_.MapWithFlags(inst, MIPSMap::NOINIT, MIPSMap::INIT | X64Map::LOW_SUBREG);` maybe.

Only a few things will need this (I think `Slt` and friends too), although I'm imagining it could be used for shifts that require CL and SSE that requires XMM0 (was hoping to avoid making that a dedicated temporary if I can avoid it, which I think is viable in IR since it already has temporaries that will get mapped.)

-[Unknown]

Reply:
Oh right, I completely forgot that movzx uses 8-bit/16-bit register subsets, and doesn't just address them as if they were full-width, which indeed is a problem on 32-bit x86...

Reply:
We could just `DISABLE;` on 32-bit.  I just know fp64 cares about 32-bit so am trying to not abandon it here.

-[Unknown]

Reply:
Yeah. I'll do that shortly